### PR TITLE
Fix ingress with v2, improve debug

### PR DESCRIPTION
**Pull request by costinm**
Note the branch is 'v2dev' - since few people are working closely on related features and we tend to 
cherry-pick each others changes: feel free to add or change to this branch, it'll be shared until we get it
to pass the master tests ( and we can start another shared branch after )!

With this in place we seem to have ingress working with LDSv2 and CDSv2 (still using RDSv1), manually
tested with 'e2e-simple' setup.

### CODE REVIEW

File: pilot/pkg/kube/inject/inject.go [hunk 1]
```diff
@@ -107,6 +107,11 @@ func InitImageName(hub string, tag string, _ bool) string {
 // ProxyImageName returns the fully qualified image name for the istio
 // proxy image given a docker hub and tag and whether to use debug or not.
 func ProxyImageName(hub string, tag string, debug bool) string {
+	// Allow overriding the proxy image.
+	image := os.Getenv("ISTIO_PROXY_IMAGE")
+	if image != "" {
+		return hub + "/" + image + ":" + tag
+	}
```
Comment (rshriram) [inline pilot/pkg/kube/inject/inject.go]: why again?

Comment (costinm) [reply pilot/pkg/kube/inject/inject.go]: It's an easy way to customize the injected image - works with both webhook and command line.

The official solution will likely be an annotation - flags in istioclt are not going to work for webhook.

**Comment by costinm** [comment]
Prow failures: seem to be happening on master, lack artifacts so not possible to find much. Passing on circle.

File: pilot/pkg/kube/inject/inject.go (hunk 1 above)
Comment (rshriram) [reply pilot/pkg/kube/inject/inject.go]: no I meant why add this? the previous PR already had this right?

**Comment by costinm** [comment]
I'm not deleting the branch: testing to see if it can be used for further changes as a 'common dev' branch to reduce number of branches and increase collaboration.

**Pull request merged**